pandas/io/formats/info.py

```diff
@@ -39,115 +41,212 @@ def _put_str(s: Union[str, Dtype], space: int) -> str:
-        %(klass)s to print information about.
-    verbose : bool, optional
-        Whether to print the full summary. By default, the setting in
-        ``pandas.options.display.max_info_columns`` is followed.
-    buf : writable buffer, defaults to sys.stdout
-        Where to send the output. By default, the output is printed to
-        sys.stdout. Pass a writable buffer if you need to further process
-        the output.
-    %(max_cols_sub)s
-    memory_usage : bool, str, optional
-        Specifies whether total memory usage of the %(klass)s
-        elements (including the index) should be displayed. By default,
-        this follows the ``pandas.options.display.memory_usage`` setting.
-
-        True always show memory usage. False never shows memory usage.
-        A value of 'deep' is equivalent to "True with deep introspection".
-        Memory usage is shown in human-readable units (base-2
-        representation). Without deep introspection a memory estimation is
-        made based in column dtype and number of rows assuming values
-        consume the same memory amount for corresponding dtypes. With deep
-        memory introspection, a real memory usage calculation is performed
-        at the cost of computational resources.
-    null_counts : bool, optional
-        Whether to show the non-null counts. By default, this is shown
-        only if the %(klass)s is smaller than
-        ``pandas.options.display.max_info_rows`` and
-        ``pandas.options.display.max_info_columns``. A value of True always
-        shows the counts, and False never shows the counts.
-
-    Returns
-    -------
-    None
-        This method prints a summary of a %(klass)s and returns None.
-
-    See Also
-    --------
-    %(see_also_sub)s
-
-    Examples
-    --------
-    %(examples_sub)s
-    """
-    if buf is None:  # pragma: no cover
-        buf = sys.stdout
-
-    lines = []
-
-    lines.append(str(type(data)))
-    lines.append(data.index._summary())
-
-    ids, dtypes = _get_ids_and_dtypes(data)
-    col_count = len(ids)
-
-    if col_count == 0:
-        lines.append(f"Empty {type(data).__name__}")
-        fmt.buffer_put_lines(buf, lines)
-        return
-
-    # hack
-    if max_cols is None:
-        max_cols = get_option("display.max_info_columns", col_count + 1)
-
-    max_rows = get_option("display.max_info_rows", len(data) + 1)
-
-    if null_counts is None:
-        show_counts = (col_count <= max_cols) and (len(data) < max_rows)
-    else:
-        show_counts = null_counts
-    exceeds_info_cols = col_count > max_cols
-
-    def _verbose_repr():
+class BaseInfo(metaclass=ABCMeta):
+    def __init__(
+        self,
+        data: FrameOrSeries,
+        verbose: Optional[bool] = None,
+        buf: Optional[IO[str]] = None,
+        max_cols: Optional[int] = None,
+        memory_usage: Optional[Union[bool, str]] = None,
+        null_counts: Optional[bool] = None,
+    ):
+        if buf is None:  # pragma: no cover
```

Comment:
why do you need the pragma here?

Reply:
Noticed this is in the current implementation. Not sure of reasoning but yea probably safe to remove

Reply:
Yes, it was from the current implementation - have removed